test/Microsoft.TemplateEngine.Mocks/MockFileSystem.cs

```diff
@@ -237,5 +237,15 @@ public DateTime GetLastWriteTimeUtc(string file)
 
             return _files[file].LastWriteTimeUtc;
         }
+
+        public void SetLastWriteTimeUtc(string file, DateTime lastWriteTimeUtc)
+        {
+            if (!FileExists(file))
```

Comment:
minor: wondering why a custom file existence method is needed instead of `System.IO.File.Exists()`

Reply:
There is a whole file system abstraction in the template engine to allow for other types beyond a physical file system.  In this case, this is a very simple in-memory file system called `MockFileSystem` so calling `System.IO.File.Exists` would not work as it would go directly to disk.